src/cdk/stepper/stepper.ts

```diff
@@ -291,13 +291,15 @@ export class CdkStepper implements OnDestroy {
     this._stepHeader.toArray()[this._focusIndex].nativeElement.focus();
   }
 
-  private _anyControlsInvalid(index: number): boolean {
+  private _anyControlsInvalidOrPending(index: number): boolean {
     const steps = this._steps.toArray();
 
     steps[this._selectedIndex].interacted = true;
 
     if (this._linear && index >= 0) {
-      return steps.slice(0, index).some(step => step.stepControl && step.stepControl.invalid);
+      return steps.slice(0, index).some(step =>
+        step.stepControl && (step.stepControl.invalid || step.stepControl.pending)
```

Comment:
can you add a unit test for a pending control?

Reply:
@mmalerba Sure. I've added unit tests and rebased the PR.